pkg/relayer/proxy/relay_verifier.go

```diff
@@ -0,0 +1,71 @@
+package proxy
+
+import (
+	"github.com/cometbft/cometbft/crypto"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	"golang.org/x/exp/slices"
+
+	"context"
+	"pocket/x/service/types"
+	sessiontypes "pocket/x/session/types"
+)
+
+// VerifyRelayRequest is a shared method used by RelayServers to check the relay request signature and session validity.
+func (rp *relayerProxy) VerifyRelayRequest(
+	ctx context.Context,
+	relayRequest *types.RelayRequest,
+	serviceId serviceId,
+) error {
+	// Query for the application account to get the application's public key to verify the relay request signature.
+	applicationAddress := relayRequest.Meta.SessionHeader.ApplicationAddress
+	accountQuery := &accounttypes.QueryAccountRequest{Address: applicationAddress}
+	accountResponse, err := rp.accountsQuerier.Account(ctx, accountQuery)
+	if err != nil {
+		return err
+	}
+
+	var payloadBz []byte
+	_, err = relayRequest.Payload.MarshalTo(payloadBz)
+	if err != nil {
+		return err
+	}
+	hash := crypto.Sha256(payloadBz)
+
+	// accountResponse.Account.Value is a protobuf Any type that should be unmarshaled into an AccountI interface.
+	// TODO_DISCUSS: Make sure this is the correct way to unmarshal an AccountI from an Any type.
```

Comment:
If you look inside `docs/static/openapi.yml`. They have this

```go
             Example 4: Pack and unpack a message in Go

                 foo := &pb.Foo{...}
                 any, err := anypb.New(foo)
                 if err != nil {
                   ...
                 }
                 ...
                 foo := &pb.Foo{}
                 if err := any.UnmarshalTo(foo); err != nil {
                   ...
                 }
```

tl;dr

1. Use the Account struct type
2. Use `any.UnmarshalTo`

Reply:
Couldn't make it work. I'll stick the current approach until we get tests or E2E

Reply:
@Olshansk we should not use `anypb.New` instead use cosmos's own `Any` type as follows:
```go
codectypes.NewAnyWithValue(value)
```